providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/common/ChannelBuilder.java

```diff
@@ -58,12 +58,12 @@ public class ChannelBuilder {
                                     "retryableStatusCodes",
                                     Arrays.asList(
                                             /*
-                                             * All codes are retryable except OK, CANCELLED and DEADLINE_EXCEEDED since
+                                             * All codes are retryable except OK and DEADLINE_EXCEEDED since
                                              * any others not listed here cause a very tight loop of retries.
-                                             * CANCELLED is not retryable because it is a client-side termination.
                                              * DEADLINE_EXCEEDED is typically a result of a client specified deadline,
                                              * and definitionally should not result in a tight loop (it's a timeout).
                                              */
+                                            Code.CANCELLED.toString(),
```

Comment:
I was hesitant to add cancelled here before because I thought it might cause delays in shutdown, but that's not the case. On channel shutdown the retries do not apply (I tested this).

It's a nice one to cover, because sometimes proto parse errors can cause `CANCELLED` to be thrown.